ruby_review_week_four.rb

```diff
@@ -1,37 +1,125 @@
 
 # 12. Define a hash called ‘student’. This hash should have two keys, one called name, and the other called age, and assign a string to the name key, and an integer to the age key. 
+# student = {
+#   name: "Melissa"
+#   age: 22
+# }
 
 # 13. Write a Ruby method that takes three arguments and returns the sum (total) of those three arguments.
+# def number_sum(num1, num2, num3)
+#   answer = num1 + num2 + num3
+# end
+
+# p number_sum(1,2,3)
 
 # 14. Write a Ruby method that takes an array and a string as arguments, and returns true if the string is in the array, and false if it is not.
+# names = ["mel", "izzy", "molly"]
+
+# def string_finder(array, string)
+#   array.include? (string)
+# end
+
+# p string_finder(names, "mel")
 
 # 15. Write a method that takes one argument, a number, and that returns true if the number is divisible by three, and false otherwise. 
 
-# 16. You must use your previous method in this next question. Write another method that takes an array (of numbers) as an argument. This method will use your first method to return two arrays (within one array - an array of arrays). The first array contains the numbers that are divisible by 3, and the other contains the rest of the numbers. For example, if you pass the second method the array [9, 4, 3, 5, 30, 2, 8] it would return [[9, 3, 30],[4, 5, 2, 8]].
\ No newline at end of file
+# def div_by_three(num1)
+#   if num1 % 3 == 0
+#     p "yes"
+#   else
+#     p 'no'
+#   end
+# end
+
+
+# 16. You must use your previous method in this next question. Write another method that takes an array (of numbers) as an argument. This method will use your first method to return two arrays (within one array - an array of arrays). The first array contains the numbers that are divisible by 3, and the other contains the rest of the numbers. For example, if you pass the second method the array [9, 4, 3, 5, 30, 2, 8] it would return [[9, 3, 30],[4, 5, 2, 8]].
+
+number_array = [2,3,6,8,11,13,18,21]
+
+def div_by_three(num1)
+  if num1 % 3 == 0
+    "yes"
+  else
+    "no"
+  end
+end
+
+def array_splitter(array)
+  threes = []
+  not_threes = []
+  array.map! do |number|
+    if 
+      div_by_three(number) == "yes"
+      threes << number
+    else 
+      div_by_three(number) == "no"
+      not_threes << number
+    end
+  end
+  array.push(threes)
+  array.push(not_threes) 
+end
+
+array_splitter(number_array)
+p number_array
+
+# I got most of the way with this. Without the exclamation mark it prints [2, 3, 6, 8, 11, 13, 18, 21, [3, 6, 18, 21], [2, 8, 11, 13]]. I dont know how to get rid of the original number without google. If i dont include the ! it prints [[2, 8, 11, 13], [3, 6, 18, 21], [3, 6, 18, 21], [2, 8, 11, 13], [2, 8, 11, 13], [2, 8, 11, 13], [3, 6, 18, 21], [3, 6, 18, 21], [3, 6, 18, 21], [2, 8, 11, 13]], which is sort of closer but the array are they too many times but the original numbers are gone. 
```

Comment:
yep you got very close here the logic is correct, if you change up the way your previous method works (div_by_three) using `return` instead of `p` it should work